wpilibc/src/main/native/cpp/smartdashboard/Mechanism2d.cpp

```diff
@@ -34,8 +34,7 @@ MechanismRoot2d* Mechanism2d::GetRoot(std::string_view name, double x,
 }
 
 void Mechanism2d::SetBackgroundColor(const Color8Bit& color) {
-  std::snprintf(m_color, sizeof(m_color), "#%02X%02X%02X", color.red,
-                color.green, color.blue);
+  m_color = color.HexString().data();
```

Comment:
This is also referencing a temporary.  Just make m_color a std::string instead.

Reply:
Does this have a corresponding Java change?

Reply:
Yes, the Mechanism2d change exists on the Java side as well.